doc/content/getting-started/api/go.md

```diff
@@ -0,0 +1,204 @@
+---
+title: "Go Code Examples"
+description: ""
+weight:
+---
+
+[Go](https://golang.org) is an open source programming language that makes it easy to build simple, reliable, and efficient software.
+
+<!-- more -->
+
+## Setup
+
+[Download](https://golang.org/dl/) and install Go. Run `go version` and make sure your version is newer than **1.15**.
+
+We will use the gRPC API of {{% tts %}}.
+
+Initialize a new Go module:
+
+```bash
+$ mkdir my_application
+$ cd my_application
+$ go mod init github.com/example/my_application
+```
+
+Add the following contents to your `go.mod` file:
+
+```go
+// go.mod
+module github.com/example/my_application
+
+go 1.15
+
+// Dependency of lorawan-stack.
+replace gopkg.in/DATA-DOG/go-sqlmock.v1 => gopkg.in/DATA-DOG/go-sqlmock.v1 v1.3.0
+
+// Dependency of lorawan-stack.
+replace gocloud.dev => gocloud.dev v0.19.0
+
+// Dependency of lorawan-stack.
+replace github.com/grpc-ecosystem/grpc-gateway => github.com/TheThingsIndustries/grpc-gateway v1.15.2-gogo
+
+require (
+    go.thethings.network/lorawan-stack/v3 v3.10.5
+    google.golang.org/grpc v1.34.0
+)
+```
+
+And then run `go mod download` to download Go dependencies.
+
+## Get End Device
+
+Create a `main.go` file and add the following code:
+
+```go
+// main.go
+package main
+
+import (
+    "context"
+    "crypto/tls"
+    "fmt"
+
+    "github.com/gogo/protobuf/types"
+    "go.thethings.network/lorawan-stack/v3/pkg/rpcmetadata"
+    "go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
+    "google.golang.org/grpc"
+    "google.golang.org/grpc/credentials"
+)
+
+var (
+    baseURL = "thethings.example.com" // Default gRPC port with TLS is 8884
+    apiKey  = rpcmetadata.MD{
+        AuthType:      "bearer",
+        AuthValue:     "NNSXS.XXXXXXXXXXXXXX.YYYYYYYYYYY",
+    }
+    userAgent = "myGoIntegration/version"
+)
+
+func main() {
+    opts := []grpc.DialOption{
+        grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{})), // require TLS
+        grpc.WithPerRPCCredentials(apiKey), // set API key for authorization
+        grpc.WithUserAgent(userAgent), // set user agent
+    }
+    cc, err := grpc.Dial(baseURL, opts...)
+    if err != nil {
+        panic(err)
+    }
+
+    dev, err := ttnpb.NewEndDeviceRegistryClient(cc).Get(context.Background(), &ttnpb.GetEndDeviceRequest{
+        EndDeviceIdentifiers: ttnpb.EndDeviceIdentifiers{
+            DeviceID:               "dev1",
+            ApplicationIdentifiers: ttnpb.ApplicationIdentifiers{ApplicationID: "app1"},
+        },
+        FieldMask: types.FieldMask{
+            Paths: []string{
+                "name",
+                "description",
+            },
+        },
+    })
+    if err != nil {
+        panic(err)
+    }
+    fmt.Println("Device id:", dev.DeviceID)
+    fmt.Println("Device name:", dev.Name)
+    fmt.Println("Device description:", dev.Description)
+}
+```
+
+Then build and run with:
```

Comment:
I think it's nice to have all this, but isn't it better if we have a (template) repository on GitHub that we maintain and that we refer to here?

I'm not sure if we should get people copy-pasting code from our docs. We can't check if it compiles, it's hard to update this, etc.

Reply:
i'm for this. we could also do what we do for installation guide, and host actual go code within the documentation if it's not going to be used anywhere else. that way at least you can easily check if it runs.

Host the files like we do the .ymls here -> https://github.com/TheThingsIndustries/lorawan-stack-docs/tree/master/doc/content/getting-started/installation/configuration

Display it in markdown like we do here -> https://github.com/TheThingsIndustries/lorawan-stack-docs/blame/master/doc/content/getting-started/installation/configuration/_index.md#L70

Reply:
Okay, so do we keep these in the documentation or create a new repository with code examples?

Reply:
The point would be to create new repos with code examples; `lorawan-stack-example-{python,go,...}`. Like https://github.com/armmbed